lib/astarte_core/mapping.ex

```diff
@@ -221,6 +222,16 @@ defmodule Astarte.Core.Mapping do
     end
   end
 
+  defp check_coherence_database_retention_policy_and_database_retention_ttl(changeset) do
+    #May return a more generic error if validate_not_set_unless/4 is used
+    if (get_field(changeset, :database_retention_policy) == :no_ttl and
+        get_field(changeset, :database_retention_ttl)  != nil) do
+        add_error(changeset, :database_retention_policy, "must be use_ttl if database_retention_ttl is set")
```

Comment:
I think this line would trigger an indentation error when passed through the CI that checks formatting (it should have 2 space indentation instead of 4). Always run `mix format` and/or setup your editor to do so when you save files.